### PR TITLE
refactor: Global renaming from hpmn to evo

### PR DESCRIPTION
**Pull request by ogabrielides**
## What was done?
Renaming of all classes/variables/functions/rpcs from `hpmn` to `evo`.

## How Has This Been Tested?
All unit and func tests are passing.
Sync of Testnet.

## Breaking Changes
All protx RPCs ending with `_hpmn` were converted to `_evo`.
`_hpmn` RPCs are now deprecated.
Although, they can still be enabled by adding `-deprecatedrpc=hpmn`.

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: test/functional/test_framework/test_framework.py [hunk 1]
```diff
@@ -1160,7 +1160,7 @@ def dynamically_prepare_masternode(self, idx, node_p2p_port, hpmn=False, rnd=Non
         platform_p2p_port = '%d' % (node_p2p_port + 101)
         platform_http_port = '%d' % (node_p2p_port + 102)
 
-        collateral_amount = 4000 if hpmn else 1000
+        collateral_amount = EVONODE_MASTERNODE_COLLATERAL if evo else MASTERNODE_COLLATERAL
```
Comment (knst) [inline test/functional/test_framework/test_framework.py:1163]: EVONODE_MASTERNODE_ sounds like node node.... May be just EVONODE_COLLATERAL ?

Comment (thephez) [reply test/functional/test_framework/test_framework.py]: I think EVOLUTION_MASTERNODE_* would work

File: src/governance/object.cpp [hunk 2]
```diff
@@ -629,8 +629,8 @@ int CGovernanceObject::CountMatchingVotes(vote_signal_enum_t eVoteSignalIn, vote
         const vote_rec_t& recVote = votepair.second;
         auto it2 = recVote.mapInstances.find(eVoteSignalIn);
         if (it2 != recVote.mapInstances.end() && it2->second.eOutcome == eVoteOutcomeIn) {
-            // 4x times weight vote for HPMN owners.
-            // No need to check if v19 is active since no HPMN are allowed to register before v19s
+            // 4x times weight vote for EvoNodes owners.
```
Comment (thephez) [inline src/governance/object.cpp:632]:
```suggestion
            // 4x times weight vote for EvoNode owners.
```

**Review by thephez — COMMENTED** [review]
Generally, I think this substitution is better than just shortening to Evo:
HighPerformance -> Evolution

But not too concerned especially for internal stuff...

**Comment by ogabrielides** [comment]
> Generally, I think this substitution is better than just shortening to Evo: HighPerformance -> Evolution
> 
> But not too concerned especially for internal stuff...

Let's gather everyone's opinion first.

**Comment by knst** [comment]
> Let's gather everyone's opinion first.

"evo" is good for rpc, because shorter to type - better.
Docs/release node/some comments/logs - Evolution is better.
Name of internal variables - evo/evolution - both ways are fine

**Comment by ogabrielides** [comment]
@UdjinM6 Your opinion ?

**Comment by UdjinM6** [comment]
I'm fine with either option.

**Comment by ogabrielides** [comment]
@PastaPastaPasta @UdjinM6 @thephez @knst 
Will implement the following then:

RPC names: `evomn`
Docs/release node/some comments/logs: `Evolution`
Internal variables: `evo`

**Comment by qwizzie** [comment]
What about the Dash Core wallet ? It still shows 'Number of HPMN's' in the Windows - Information tab.

**Comment by UdjinM6** [comment]
LGTM overall, pls see d16b7ee528 for a couple of small suggestions in comments

File: doc/release-notes-5508.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+Updated RPCs
+--------
+
+- `protx register_hpmn` is renamed to `protx register_evo`.
```
[the remark this answers is not in the document]
Comment (ogabrielides) [reply doc/release-notes-5508.md:4]: @PastaPastaPasta Well.. we say EvoNode not EvoMasternode so..

**Review by UdjinM6 — APPROVED** [review]
utACK

File: src/rpc/evo.cpp [hunk 4]
```diff
@@ -1616,16 +1616,16 @@ static UniValue protx(const JSONRPCRequest& request)
 #ifdef ENABLE_WALLET
     if (command == "protxregister" || command == "protxregister_fund" || command == "protxregister_prepare") {
```
Comment (knst) [inline src/rpc/evo.cpp:1617]: it seems as there's no way to warn user if he use deprecated RPC such as any message/etc, or does a way exist?

```
    if (command.size() > 5 && command.substr(command.size() - 5) == "_hpmn") {
        rpc_warning("Deprecated API! Use _evo suffix instead _hpmn");
    }

**Review by knst — APPROVED** [review]
utACK.

See a comment also.

File: src/rpc/evo.cpp (hunk 4 above)
Comment (UdjinM6) [reply src/rpc/evo.cpp:1617]: Good idea! We can use `IsDeprecatedRPCEnabled` for this I guess. Also, I think `_hpmn` commands are actually broken now. So we need smth like
```diff
diff --git a/src/rpc/evo.cpp b/src/rpc/evo.cpp
index c8c8f9f37c..71897b5d86 100644
--- a/src/rpc/evo.cpp
+++ b/src/rpc/evo.cpp
@@ -793,6 +793,14 @@ static UniValue protx_register_evo(const JSONRPCRequest& request, const Chainsta
     bool isExternalRegister = request.strMethod == "protxregister_evo";
     bool isFundRegister = request.strMethod == "protxregister_fund_evo";
     bool isPrepareRegister = request.strMethod == "protxregister_prepare_evo";
+    if (request.strMethod.find("_hpmn") != std::string::npos) {
+        if (!IsDeprecatedRPCEnabled("hpmn")) {
+            throw JSONRPCError(RPC_METHOD_DEPRECATED, "*_hpmn methods are deprecated, set -deprecatedrpc=hpmn to enable them");
+        }
+        isExternalRegister = request.strMethod == "protxregister_hpmn";
+        isFundRegister = request.strMethod == "protxregister_fund_hpmn";
+        isPrepareRegister = request.strMethod == "protxregister_prepare_hpmn";
+    }
     return protx_register_common_wrapper(request, chainman, false, isExternalRegister, isFundRegister, isPrepareRegister, MnType::Evo);
 }
 
```

**Review by knst — DISMISSED** [review]
utACK

**Review by knst — DISMISSED** [review]
re-utACK

File: src/rpc/evo.cpp [hunk 5]
```diff
@@ -788,12 +788,20 @@ static UniValue protx_register_common_wrapper(const JSONRPCRequest& request,
...
-    bool isExternalRegister = request.strMethod == "protxregister_hpmn";
-    bool isFundRegister = request.strMethod == "protxregister_fund_hpmn";
-    bool isPrepareRegister = request.strMethod == "protxregister_prepare_hpmn";
-    return protx_register_common_wrapper(request, chainman, false, isExternalRegister, isFundRegister, isPrepareRegister, MnType::HighPerformance);
+    bool isExternalRegister = request.strMethod == "protxregister_evo";
+    bool isFundRegister = request.strMethod == "protxregister_fund_evo";
+    bool isPrepareRegister = request.strMethod == "protxregister_prepare_evo";
+    if (request.strMethod.find("_hpmn") != std::string::npos) {
+        if (!IsDeprecatedRPCEnabled("hpmn")) {
+            throw JSONRPCError(RPC_METHOD_DEPRECATED, "*_hpmn methods are deprecated, set -deprecatedrpc=hpmn to enable them");
```
[the remark this answers is not in the document]
Comment (ogabrielides) [reply src/rpc/evo.cpp:798]: @thephez Valid point!

**Review by thephez — APPROVED** [review]
utACK

**Review by knst — DISMISSED** [review]
re-utACK

**Review by UdjinM6 — DISMISSED** [review]
utACK

**Review by thephez — APPROVED** [review]
re-utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**